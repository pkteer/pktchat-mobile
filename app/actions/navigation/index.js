// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import merge from 'deepmerge';
import {Keyboard, Platform} from 'react-native';
import {Navigation} from 'react-native-navigation';

import CompassIcon from '@components/compass_icon';
import {DeviceTypes, NavigationTypes} from '@constants';
import {CHANNEL} from '@constants/screen';
import {Preferences} from '@mm-redux/constants';
import {getTheme} from '@mm-redux/selectors/entities/preferences';
import EventEmmiter from '@mm-redux/utils/event_emitter';
import EphemeralStore from '@store/ephemeral_store';
import Store from '@store/store';

Navigation.setDefaultOptions({
    layout: {
        orientation: [DeviceTypes.IS_TABLET ? 'all' : 'portrait'],
    },
});

function getThemeFromState() {
    const state = Store.redux?.getState() || {};

    return getTheme(state);
}

export function resetToChannel(passProps = {}) {
    const theme = getThemeFromState();

    EphemeralStore.clearNavigationComponents();

    const stack = {
        children: [{
            component: {
                id: CHANNEL,
                name: CHANNEL,
                passProps,
                options: {
                    layout: {
                        componentBackgroundColor: theme.centerChannelBg,
                    },
                    statusBar: {
                        visible: true,
                    },
                    topBar: {
                        visible: false,
                        height: 0,
                        background: {
                            color: theme.sidebarHeaderBg,
                        },
                        backButton: {
                            visible: false,
                            color: theme.sidebarHeaderTextColor,
                            enableMenu: false,
                        },
                    },
                },
            },
        }],
    };

    let platformStack = {stack};
    if (Platform.OS === 'android') {
        platformStack = {
            sideMenu: {
                left: {
                    component: {
                        id: 'MainSidebar',
                        name: 'MainSidebar',
                    },
                },
                center: {
                    stack,
                },
                right: {
                    component: {
                        id: 'SettingsSidebar',
                        name: 'SettingsSidebar',
                    },
                },
            },
        };
    }

    Navigation.setRoot({
        root: {
            ...platformStack,
        },
    });
}

export function resetToSelectServer(allowOtherServers) {
    const theme = Preferences.THEMES.denim;

    EphemeralStore.clearNavigationComponents();

    Navigation.setRoot({
        root: {
            stack: {
                children: [{
                    component: {
                        id: 'SelectServer',
                        name: 'SelectServer',
                        passProps: {
                            allowOtherServers,
                        },
                        options: {
                            layout: {
                                backgroundColor: theme.centerChannelBg,
                                componentBackgroundColor: theme.centerChannelBg,
                            },
                            statusBar: {
                                visible: true,
                            },
                            topBar: {
                                backButton: {
                                    color: theme.sidebarHeaderTextColor,
                                    enableMenu: false,
                                    title: '',
                                },
                                background: {
                                    color: theme.sidebarHeaderBg,
                                },
                                visible: false,
                                height: 0,
                            },
                        },
                    },
                }],
            },
        },
    });
}

export function resetToTeams(name, title, passProps = {}, options = {}) {
    const theme = getThemeFromState();
    const defaultOptions = {
        layout: {
            componentBackgroundColor: theme.centerChannelBg,
        },
        statusBar: {
            visible: true,
        },
        topBar: {
            visible: true,
            title: {
                color: theme.sidebarHeaderTextColor,
                text: title,
            },
            backButton: {
                color: theme.sidebarHeaderTextColor,
                enableMenu: false,
                title: '',
            },
            background: {
                color: theme.sidebarHeaderBg,
            },
        },
    };

    EphemeralStore.clearNavigationComponents();

    Navigation.setRoot({
        root: {
            stack: {
                children: [{
                    component: {
                        id: name,
                        name,
                        passProps,
                        options: merge(defaultOptions, options),
                    },
                }],
            },
        },
    });
}

export function goToScreen(name, title, passProps = {}, options = {}) {
    const theme = getThemeFromState();
    const componentId = EphemeralStore.getNavigationTopComponentId();
    const defaultOptions = {
        layout: {
            componentBackgroundColor: theme.centerChannelBg,
        },
        popGesture: true,
        sideMenu: {
            left: {enabled: false},
            right: {enabled: false},
        },
        topBar: {
            animate: true,
            visible: true,
            backButton: {
                color: theme.sidebarHeaderTextColor,
                enableMenu: false,
                title: '',
                testID: 'screen.back.button',
            },
            background: {
                color: theme.sidebarHeaderBg,
            },
            title: {
                color: theme.sidebarHeaderTextColor,
                text: title,
            },
        },
    };

    Navigation.push(componentId, {
        component: {
            id: name,
            name,
            passProps,
            options: merge(defaultOptions, options),
        },
    });
}

export function popTopScreen(screenId) {
    if (screenId) {
        Navigation.pop(screenId);
    } else {
        const componentId = EphemeralStore.getNavigationTopComponentId();
        Navigation.pop(componentId);
    }
}

export async function popToRoot() {
    const componentId = EphemeralStore.getNavigationTopComponentId();

    try {
        await Navigation.popToRoot(componentId);
    } catch (error) {
        // RNN returns a promise rejection if there are no screens
        // atop the root screen to pop. We'll do nothing in this case.
    }
}

export async function dismissAllModalsAndPopToRoot() {
    await dismissAllModals();
    await popToRoot();

    EventEmmiter.emit(NavigationTypes.NAVIGATION_DISMISS_AND_POP_TO_ROOT);
}

export function showModal(name, title, passProps = {}, options = {}) {
    const theme = getThemeFromState();
    const defaultOptions = {
        modalPresentationStyle: Platform.select({ios: 'pageSheet', android: 'none'}),
        layout: {
            componentBackgroundColor: theme.centerChannelBg,
        },
        statusBar: {
            visible: true,
        },
        topBar: {
            animate: true,
            visible: true,
            backButton: {
                color: theme.sidebarHeaderTextColor,
                enableMenu: false,
                title: '',
            },
            background: {
                color: theme.sidebarHeaderBg,
            },
            title: {
                color: theme.sidebarHeaderTextColor,
                text: title,
            },
            leftButtonColor: theme.sidebarHeaderTextColor,
            rightButtonColor: theme.sidebarHeaderTextColor,
        },
    };

    EphemeralStore.addNavigationModal(name);
    Navigation.showModal({
        stack: {
            children: [{
                component: {
                    id: name,
                    name,
                    passProps,
                    options: merge(defaultOptions, options),
                },
            }],
        },
    });
}

export function showModalOverCurrentContext(name, passProps = {}, options = {}) {
    const title = passProps.title || '';

    let animations;
    switch (Platform.OS) {
    case 'android':
        animations = {
            showModal: {
                waitForRender: true,
                alpha: {
                    from: 0,
                    to: 1,
                    duration: 250,
                },
            },
            dismissModal: {
                alpha: {
                    from: 1,
                    to: 0,
                    duration: 250,
                },
            },
        };
        break;
    default:
        animations = {
            showModal: {
                enter: {
                    enabled: false,
                },
                exit: {
                    enabled: false,
                },
            },
            dismissModal: {
                enter: {
                    enabled: false,
                },
                exit: {
                    enabled: false,
                },
            },
        };
        break;
    }

    const defaultOptions = {
        modalPresentationStyle: 'overCurrentContext',
        layout: {
            backgroundColor: 'transparent',
            componentBackgroundColor: 'transparent',
        },
        topBar: {
            visible: false,
            height: 0,
        },
        animations,
    };
    const mergeOptions = merge(defaultOptions, options);

    showModal(name, title, passProps, mergeOptions);
}

export function showSearchModal(initialValue = '') {
    const name = 'Search';
    const title = '';
    const passProps = {initialValue};
    const options = {
        topBar: {
            visible: false,
            height: 0,
        },
        ...Platform.select({
            ios: {
                modalPresentationStyle: 'pageSheet',
            },
        }),
    };

    showModal(name, title, passProps, options);
}

export const showAppForm = async (form, context, theme) => {
    const closeButton = await CompassIcon.getImageSource('close', 24, theme.sidebarHeaderTextColor);

    let submitButtons;
    const customSubmitButtons = form.submit_buttons && form.fields.find((f) => f.name === form.submit_buttons)?.options;

    if (!customSubmitButtons?.length) {
        submitButtons = [{
            id: 'submit-form',
            showAsAction: 'always',
            text: 'Submit',
        }];
    }

    const options = {
        topBar: {
            leftButtons: [{
                id: 'close-dialog',
                icon: closeButton,
            }],
            rightButtons: submitButtons,
        },
    };

    const passProps = {form, context};
    showModal('AppForm', form.title, passProps, options);
};

export async function dismissModal(options = {}) {
    if (!EphemeralStore.hasModalsOpened()) {
        return;
    }

    const componentId = options.componentId || EphemeralStore.getNavigationTopComponentId();

    try {
        await Navigation.dismissModal(componentId, options);
        EphemeralStore.removeNavigationModal(componentId);
    } catch (error) {
        // RNN returns a promise rejection if there is no modal to
        // dismiss. We'll do nothing in this case.
    }
}

export async function dismissAllModals(options) {
    if (!EphemeralStore.hasModalsOpened()) {
        return;
    }

    if (Platform.OS === 'ios') {
        const modals = [...EphemeralStore.navigationModalStack];
        for await (const modal of modals) {
            await Navigation.dismissModal(modal, options);
            EphemeralStore.removeNavigationModal(modal);
        }
    } else {
        await Navigation.dismissAllModals(options);
        EphemeralStore.clearNavigationModals();
    }
}

export function setButtons(componentId, buttons = {leftButtons: [], rightButtons: []}) {
    const options = {
        topBar: {
            ...buttons,
        },
    };

    mergeNavigationOptions(componentId, options);
}

export function mergeNavigationOptions(componentId, options) {
    Navigation.mergeOptions(componentId, options);
}

export function showOverlay(name, passProps, options = {}) {
    const defaultOptions = {
        layout: {
            backgroundColor: 'transparent',
            componentBackgroundColor: 'transparent',
        },
        overlay: {
            interceptTouchOutside: false,
        },
        ...Platform.select({
            android: {
                statusBar: {
                    drawBehind: true,
                },
            },
        }),
    };

    Navigation.showOverlay({
        component: {
            name,
            passProps,
            options: merge(defaultOptions, options),
        },
    });
}

export async function dismissOverlay(componentId) {
    try {
        await Navigation.dismissOverlay(componentId);
    } catch (error) {
        // RNN returns a promise rejection if there is no modal with
        // this componentId to dismiss. We'll do nothing in this case.
    }
}

export function openMainSideMenu() {
    if (Platform.OS === 'ios') {
        return;
    }

    const componentId = EphemeralStore.getNavigationTopComponentId();

    Keyboard.dismiss();
    Navigation.mergeOptions(componentId, {
        sideMenu: {
            left: {visible: true},
        },
    });
}

export function closeMainSideMenu() {
    if (Platform.OS === 'ios') {
        return;
    }

    Keyboard.dismiss();
    Navigation.mergeOptions(CHANNEL, {
        sideMenu: {
            left: {visible: false},
        },
    });
}

export function enableMainSideMenu(enabled, visible = true) {
    if (Platform.OS === 'ios') {
        return;
    }

    Navigation.mergeOptions(CHANNEL, {
        sideMenu: {
            left: {enabled, visible},
        },
    });
}

export function openSettingsSideMenu() {
    if (Platform.OS === 'ios') {
        return;
    }

    Keyboard.dismiss();
    Navigation.mergeOptions(CHANNEL, {
        sideMenu: {
            right: {visible: true},
        },
    });
}

export function closeSettingsSideMenu() {
    if (Platform.OS === 'ios') {
        return;
    }

    Keyboard.dismiss();
    Navigation.mergeOptions(CHANNEL, {
        sideMenu: {
            right: {visible: false},
        },
    });
}
