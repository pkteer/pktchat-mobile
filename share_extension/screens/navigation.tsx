// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';

import React from 'react';

import {Preferences} from '@mm-redux/constants';

import Channels from './channel_list';
import Share from './share';
import Teams from './team_list';

const theme = Preferences.THEMES.denim;

const Stack = createStackNavigator();
const defaultNavigationOptions = {
    headerStyle: {
        backgroundColor: theme.sidebarHeaderBg,
    },
    headerTitleStyle: {
        marginHorizontal: 0,
        left: 0,
        color: theme.sidebarHeaderTextColor,
    },
    headerBackTitleStyle: {
        color: theme.sidebarHeaderTextColor,
        margin: 0,
    },
    headerTintColor: theme.sidebarHeaderTextColor,
    headerTopInsetEnabled: false,
};

function RootStack() {
    return (
        <Stack.Navigator
            initialRouteName='Post'
            screenOptions={defaultNavigationOptions}
        >
            <Stack.Screen
                name='Post'
                component={Share}
            />
            <Stack.Screen
                name='Teams'
                component={Teams}
            />
            <Stack.Screen
                name='Channels'
                component={Channels}
            />
        </Stack.Navigator>
    );
}

export default function ExtensionNavigation() {
    return (
        <NavigationContainer>
            <RootStack/>
        </NavigationContainer>
    );
}
