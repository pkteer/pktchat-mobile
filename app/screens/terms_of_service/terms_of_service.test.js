// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import * as NavigationActions from '@actions/navigation';
import Preferences from '@mm-redux/constants/preferences';
import {shallowWithIntl} from '@test/intl-test-helper';
import TestHelper from '@test/test_helper';

import TermsOfService from './terms_of_service.js';

jest.mock('@utils/theme', () => {
    const original = jest.requireActual('../../utils/theme');
    return {
        ...original,
        changeOpacity: jest.fn(),
    };
});

describe('TermsOfService', () => {
    const actions = {
        getTermsOfService: jest.fn(),
        updateMyTermsOfServiceStatus: jest.fn().mockResolvedValue({data: true}),
        logout: jest.fn(),
    };

    const baseProps = {
        actions,
        closeButton: {},
        componentId: 'component-id',
        isSupportedServer: true,
        siteName: 'Mattermost',
        showUnsupportedServer: jest.fn(),
        theme: Preferences.THEMES.denim,
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <TermsOfService {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for get terms', async () => {
        const wrapper = shallowWithIntl(
            <TermsOfService {...baseProps}/>,
        );
        await actions.getTermsOfService();
        wrapper.update();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for fail of get terms', async () => {
        const getTermsOfService = async () => {
            return {
                error: {},
            };
        };

        const props = {
            ...baseProps,
            actions: {
                ...actions,
                getTermsOfService,
            },
        };

        const wrapper = shallowWithIntl(
            <TermsOfService {...props}/>,
        );
        expect(wrapper.state('loading')).toEqual(true);
        await getTermsOfService();
        expect(wrapper.state('loading')).toEqual(false);
        wrapper.update();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should call setButtons on setNavigatorButtons', async () => {
        const setButtons = jest.spyOn(NavigationActions, 'setButtons');

        const wrapper = shallowWithIntl(
            <TermsOfService {...baseProps}/>,
        );
        wrapper.setState({loading: false, termsId: 1, termsText: 'Terms Text'});

        expect(setButtons).toHaveBeenCalledTimes(2);
        wrapper.instance().setNavigatorButtons(true);
        expect(setButtons).toHaveBeenCalledTimes(3);
        wrapper.instance().setNavigatorButtons(false);
        expect(setButtons).toHaveBeenCalledTimes(4);
    });

    test('should enable/disable navigator buttons on setNavigatorButtons true/false', () => {
        const wrapper = shallowWithIntl(
            <TermsOfService {...baseProps}/>,
        );
        wrapper.setState({loading: false, termsId: 1, termsText: 'Terms Text'});

        wrapper.instance().setNavigatorButtons(true);
        expect(wrapper.getElement()).toMatchSnapshot();
        wrapper.instance().setNavigatorButtons(false);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot on enableNavigatorLogout', () => {
        const wrapper = shallowWithIntl(
            <TermsOfService {...baseProps}/>,
        );

        wrapper.setState({loading: false, termsId: 1, termsText: 'Terms Text'});
        wrapper.instance().enableNavigatorLogout();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should call dismissAllModals on closeTermsAndLogout', () => {
        const wrapper = shallowWithIntl(
            <TermsOfService {...baseProps}/>,
        );

        wrapper.setState({loading: false, termsId: 1, termsText: 'Terms Text'});
        wrapper.instance().closeTermsAndLogout();
        expect(baseProps.actions.logout).toHaveBeenCalledTimes(1);
    });

    test('should NOT call showUnsupportedServer on acceptTerms if server is supported', async () => {
        const wrapper = shallowWithIntl(
            <TermsOfService
                {...baseProps}
            />,
        );

        wrapper.setState({loading: false, termsId: 1, termsText: 'Terms Text'});
        wrapper.instance().handleAcceptTerms();
        TestHelper.wait(100).then(() => {
            expect(baseProps.showUnsupportedServer).not.toHaveBeenCalled();
        });
    });

    test('should call showUnsupportedServer on acceptTerms if server not supported', async () => {
        const wrapper = shallowWithIntl(
            <TermsOfService
                {...baseProps}
                isSupportedServer={false}
            />,
        );

        wrapper.setState({loading: false, termsId: 1, termsText: 'Terms Text'});
        wrapper.instance().handleAcceptTerms();
        TestHelper.wait(100).then(() => {
            expect(baseProps.showUnsupportedServer).toHaveBeenCalledTimes(1);
        });
    });
});
