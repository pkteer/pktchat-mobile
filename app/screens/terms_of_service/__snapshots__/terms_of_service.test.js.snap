// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsOfService should enable/disable navigator buttons on setNavigatorButtons true/false 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <ScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 50,
      }
    }
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "padding": 30,
      }
    }
  >
    <Connect(Markdown)
      baseTextStyle={
        Object {
          "color": "#3f4350",
          "fontSize": 15,
          "lineHeight": 20,
        }
      }
      blockStyles={
        Object {
          "adjacentParagraph": Object {
            "marginTop": 6,
          },
          "horizontalRule": Object {
            "backgroundColor": "#3f4350",
            "height": 0.5,
            "marginVertical": 10,
          },
          "quoteBlockIcon": Object {
            "color": undefined,
          },
        }
      }
      disableAtMentions={true}
      disableChannelLink={true}
      disableGallery={true}
      disableHashtags={true}
      textStyles={
        Object {
          "code": Object {
            "alignSelf": "center",
            "backgroundColor": undefined,
            "fontFamily": "Menlo",
          },
          "codeBlock": Object {
            "fontFamily": "Menlo",
          },
          "del": Object {
            "textDecorationLine": "line-through",
          },
          "emph": Object {
            "fontStyle": "italic",
          },
          "error": Object {
            "color": "#d24b4e",
          },
          "heading1": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading1Text": Object {
            "paddingBottom": 8,
          },
          "heading2": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading2Text": Object {
            "paddingBottom": 8,
          },
          "heading3": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading3Text": Object {
            "paddingBottom": 8,
          },
          "heading4": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading4Text": Object {
            "paddingBottom": 8,
          },
          "heading5": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading5Text": Object {
            "paddingBottom": 8,
          },
          "heading6": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading6Text": Object {
            "paddingBottom": 8,
          },
          "link": Object {
            "color": "#386fe5",
          },
          "mention": Object {
            "color": "#386fe5",
          },
          "mention_highlight": Object {
            "backgroundColor": "#ffd470",
            "color": "#1b1d22",
          },
          "strong": Object {
            "fontWeight": "bold",
          },
          "table_header_row": Object {
            "fontWeight": "700",
          },
        }
      }
      value="Terms Text"
    />
  </ScrollView>
</RNCSafeAreaView>
`;

exports[`TermsOfService should enable/disable navigator buttons on setNavigatorButtons true/false 2`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <ScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 50,
      }
    }
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "padding": 30,
      }
    }
  >
    <Connect(Markdown)
      baseTextStyle={
        Object {
          "color": "#3f4350",
          "fontSize": 15,
          "lineHeight": 20,
        }
      }
      blockStyles={
        Object {
          "adjacentParagraph": Object {
            "marginTop": 6,
          },
          "horizontalRule": Object {
            "backgroundColor": "#3f4350",
            "height": 0.5,
            "marginVertical": 10,
          },
          "quoteBlockIcon": Object {
            "color": undefined,
          },
        }
      }
      disableAtMentions={true}
      disableChannelLink={true}
      disableGallery={true}
      disableHashtags={true}
      textStyles={
        Object {
          "code": Object {
            "alignSelf": "center",
            "backgroundColor": undefined,
            "fontFamily": "Menlo",
          },
          "codeBlock": Object {
            "fontFamily": "Menlo",
          },
          "del": Object {
            "textDecorationLine": "line-through",
          },
          "emph": Object {
            "fontStyle": "italic",
          },
          "error": Object {
            "color": "#d24b4e",
          },
          "heading1": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading1Text": Object {
            "paddingBottom": 8,
          },
          "heading2": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading2Text": Object {
            "paddingBottom": 8,
          },
          "heading3": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading3Text": Object {
            "paddingBottom": 8,
          },
          "heading4": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading4Text": Object {
            "paddingBottom": 8,
          },
          "heading5": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading5Text": Object {
            "paddingBottom": 8,
          },
          "heading6": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading6Text": Object {
            "paddingBottom": 8,
          },
          "link": Object {
            "color": "#386fe5",
          },
          "mention": Object {
            "color": "#386fe5",
          },
          "mention_highlight": Object {
            "backgroundColor": "#ffd470",
            "color": "#1b1d22",
          },
          "strong": Object {
            "fontWeight": "bold",
          },
          "table_header_row": Object {
            "fontWeight": "700",
          },
        }
      }
      value="Terms Text"
    />
  </ScrollView>
</RNCSafeAreaView>
`;

exports[`TermsOfService should match snapshot 1`] = `
<Loading
  color="#3f4350"
  size="large"
  style={Object {}}
/>
`;

exports[`TermsOfService should match snapshot for fail of get terms 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <Connect(FailedNetworkAction)
    onRetry={[Function]}
    theme={
      Object {
        "awayIndicator": "#ffbc1f",
        "buttonBg": "#1c58d9",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3f4350",
        "codeTheme": "github",
        "dndIndicator": "#d24b4e",
        "errorTextColor": "#d24b4e",
        "linkColor": "#386fe5",
        "mentionBg": "#ffffff",
        "mentionColor": "#1e325c",
        "mentionHighlightBg": "#ffd470",
        "mentionHighlightLink": "#1b1d22",
        "newMessageSeparator": "#cc8f00",
        "onlineIndicator": "#3db887",
        "sidebarBg": "#1e325c",
        "sidebarHeaderBg": "#192a4d",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarTeamBarBg": "#14213e",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#5d89ea",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#28427b",
        "sidebarUnreadText": "#ffffff",
        "type": "Denim",
      }
    }
  />
</View>
`;

exports[`TermsOfService should match snapshot for get terms 1`] = `
<Loading
  color="#3f4350"
  size="large"
  style={Object {}}
/>
`;

exports[`TermsOfService should match snapshot on enableNavigatorLogout 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <ScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 50,
      }
    }
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "padding": 30,
      }
    }
  >
    <Connect(Markdown)
      baseTextStyle={
        Object {
          "color": "#3f4350",
          "fontSize": 15,
          "lineHeight": 20,
        }
      }
      blockStyles={
        Object {
          "adjacentParagraph": Object {
            "marginTop": 6,
          },
          "horizontalRule": Object {
            "backgroundColor": "#3f4350",
            "height": 0.5,
            "marginVertical": 10,
          },
          "quoteBlockIcon": Object {
            "color": undefined,
          },
        }
      }
      disableAtMentions={true}
      disableChannelLink={true}
      disableGallery={true}
      disableHashtags={true}
      textStyles={
        Object {
          "code": Object {
            "alignSelf": "center",
            "backgroundColor": undefined,
            "fontFamily": "Menlo",
          },
          "codeBlock": Object {
            "fontFamily": "Menlo",
          },
          "del": Object {
            "textDecorationLine": "line-through",
          },
          "emph": Object {
            "fontStyle": "italic",
          },
          "error": Object {
            "color": "#d24b4e",
          },
          "heading1": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading1Text": Object {
            "paddingBottom": 8,
          },
          "heading2": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading2Text": Object {
            "paddingBottom": 8,
          },
          "heading3": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading3Text": Object {
            "paddingBottom": 8,
          },
          "heading4": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading4Text": Object {
            "paddingBottom": 8,
          },
          "heading5": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading5Text": Object {
            "paddingBottom": 8,
          },
          "heading6": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading6Text": Object {
            "paddingBottom": 8,
          },
          "link": Object {
            "color": "#386fe5",
          },
          "mention": Object {
            "color": "#386fe5",
          },
          "mention_highlight": Object {
            "backgroundColor": "#ffd470",
            "color": "#1b1d22",
          },
          "strong": Object {
            "fontWeight": "bold",
          },
          "table_header_row": Object {
            "fontWeight": "700",
          },
        }
      }
      value="Terms Text"
    />
  </ScrollView>
</RNCSafeAreaView>
`;
